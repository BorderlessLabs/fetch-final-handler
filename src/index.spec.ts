import { Request } from "cross-fetch";
import { finalHandler } from "./index";

describe("fetch final handler", () => {
  const handler = finalHandler();

  it("should return a 404 response", async () => {
    const req = new Request("/");
    const res = await handler(req);
    expect(res).toMatchSnapshot();
  });

  it("should not return a body when using HEAD", async () => {
    const req = new Request("/", { method: "head" });
    const res = await handler(req);
    expect(res).toMatchSnapshot();
  });
});
