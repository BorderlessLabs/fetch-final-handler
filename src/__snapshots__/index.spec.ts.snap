// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch final handler should not return a body when using HEAD 1`] = `
Response {
  "size": 0,
  "timeout": 0,
  Symbol(Body internals): Object {
    "body": null,
    "disturbed": false,
    "error": null,
  },
  Symbol(Response internals): Object {
    "counter": undefined,
    "headers": Headers {
      Symbol(map): Object {
        "Content-Security-Policy": Array [
          "default-src 'self'",
        ],
        "X-Content-Type-Options": Array [
          "nosniff",
        ],
      },
    },
    "status": 404,
    "statusText": "Not Found",
    "url": undefined,
  },
}
`;

exports[`fetch final handler should return a 404 response 1`] = `
Response {
  "size": 0,
  "timeout": 0,
  Symbol(Body internals): Object {
    "body": Object {
      "data": Array [
        67,
        97,
        110,
        110,
        111,
        116,
        32,
        71,
        69,
        84,
        32,
        47,
      ],
      "type": "Buffer",
    },
    "disturbed": false,
    "error": null,
  },
  Symbol(Response internals): Object {
    "counter": undefined,
    "headers": Headers {
      Symbol(map): Object {
        "Content-Security-Policy": Array [
          "default-src 'self'",
        ],
        "Content-Type": Array [
          "text/plain;charset=UTF-8",
        ],
        "X-Content-Type-Options": Array [
          "nosniff",
        ],
      },
    },
    "status": 404,
    "statusText": "Not Found",
    "url": undefined,
  },
}
`;
